### PR TITLE
Run FindBugs in CI

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java [hunk 1]
```diff
@@ -855,7 +855,7 @@ private static BigDecimal bigSqrt(BigDecimal squarD, MathContext rootMC) {
             // The loop of "Square Root by Coupled Newton Iteration"
             for (int i = nPrecs.size() - 1; i > -1; i--) {
                 // Increase precision - next iteration supplies n exact digits
-                nMC = new MathContext(nPrecs.get(i), i%2 == 1 ? RoundingMode.HALF_UP : RoundingMode.HALF_DOWN);
+                nMC = new MathContext(nPrecs.get(i), i % 2 != 0 ? RoundingMode.HALF_UP : RoundingMode.HALF_DOWN);
```
Comment (eregon) [inline src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java]: @pitr-ch I am not sure if this change and the one above is correct.
The new check now checks whether i is *odd*.
The old check was reported by findbugs:
```
M D IM: Check for oddness that won't work for negative numbers
```
See http://findbugs.sourceforge.net/bugDescriptions.html#IM_BAD_CHECK_FOR_ODD
`-5 % 2 => -1`
So, do you think this code is meant to check oddness or only for positive numbers `i > 0 && i % 2 == 1` ?

Comment (pitr-ch) [reply src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java]: `i` is never negative, the change is correct.  Please add assert for `scale` being positive, I did not find any path leading to it being negative, but it's much more complex I could have missed something.

Comment (eregon) [reply src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java]: `scale` can have negative values according to its JavaDoc. But maybe we don't build a BigDecimal with a negative scale?

Comment (eregon) [reply src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java]: I added the `assert`.

Comment (pitr-ch) [reply src/main/java/org/truffleruby/stdlib/bigdecimal/BigDecimalNodes.java]: yeah exactly I did not find any place where we could build it with negative number or any operation on BigDecimal which could result in negative scale.

**Pull request merged**